std/algebra/emulated/fields_bn254/e6.go

```diff
@@ -0,0 +1,211 @@
+	t5 := e.Ext2.Mul(&x.B1, &x.B2)   // t5.Mul(&x.B1, &x.B2)
+	c0 := e.Ext2.MulByNonResidue(t5) // c0.MulByNonResidue(&t5).
+	c0 = e.Ext2.Neg(c0)              //    Neg(&c0).
+	c0 = e.Ext2.Add(c0, t0)          //    Add(&c0, &t0)
+	c1 := e.Ext2.MulByNonResidue(t2) // c1.MulByNonResidue(&t2).
+	c1 = e.Ext2.Sub(c1, t3)          //    Sub(&c1, &t3)
+	c2 := e.Ext2.Sub(t1, t4)         // c2.Sub(&t1, &t4)
+	t6 := e.Ext2.Mul(&x.B0, c0)      // t6.Mul(&x.B0, &c0)
+	d1 := e.Ext2.Mul(&x.B2, c1)      // d1.Mul(&x.B2, &c1)
+	d2 := e.Ext2.Mul(&x.B1, c2)      // d2.Mul(&x.B1, &c2)
+	d1 = e.Ext2.Add(d1, d2)          // d1.Add(&d1, &d2).
+	d1 = e.Ext2.MulByNonResidue(d1)  //    MulByNonResidue(&d1)
+	t6 = e.Ext2.Add(t6, d1)          // t6.Add(&t6, &d1)
+	t6 = e.Ext2.Inverse(t6)          // t6.Inverse(&t6)
+	z0 := e.Ext2.Mul(c0, t6)         // z.B0.Mul(&c0, &t6)
+	z1 := e.Ext2.Mul(c1, t6)         // z.B1.Mul(&c1, &t6)
+	z2 := e.Ext2.Mul(c2, t6)         // z.B2.Mul(&c2, &t6)
+	return &E6{                      // return z
+		B0: *z0,
+		B1: *z1,
+		B2: *z2,
+	}
+}
+func (e Ext6) MulByE2(x *E6, y *E2) *E6 {
+	// var yCopy E2
+	// yCopy.Set(y)
+	z0 := e.Ext2.Mul(&x.B0, y) // z.B0.Mul(&x.B0, &yCopy)
+	z1 := e.Ext2.Mul(&x.B1, y) // z.B1.Mul(&x.B1, &yCopy)
+	z2 := e.Ext2.Mul(&x.B2, y) // z.B2.Mul(&x.B2, &yCopy)
+	return &E6{                // return z
+		B0: *z0,
+		B1: *z1,
+		B2: *z2,
+	}
+}
+
+func (e Ext6) MulBy01(z *E6, c0, c1 *E2) *E6 {
+	// var a, b, tmp, t0, t1, t2 E2
+	a := e.Ext2.Mul(&z.B0, c0)      // a.Mul(&z.B0, c0)
+	b := e.Ext2.Mul(&z.B1, c1)      // b.Mul(&z.B1, c1)
+	tmp := e.Ext2.Add(&z.B1, &z.B2) // tmp.Add(&z.B1, &z.B2)
+	t0 := e.Ext2.Mul(c1, tmp)       // t0.Mul(c1, &tmp)
+	t0 = e.Ext2.Sub(t0, b)          // t0.Sub(&t0, &b)
+	t0 = e.Ext2.MulByNonResidue(t0) // t0.MulByNonResidue(&t0)
+	t0 = e.Ext2.Add(t0, a)          // t0.Add(&t0, &a)
+	tmp = e.Ext2.Add(&z.B0, &z.B2)  // tmp.Add(&z.B0, &z.B2)
+	t2 := e.Ext2.Mul(c0, tmp)       // t2.Mul(c0, &tmp)
+	t2 = e.Ext2.Sub(t2, a)          // t2.Sub(&t2, &a)
+	t2 = e.Ext2.Add(t2, b)          // t2.Add(&t2, &b)
+	t1 := e.Ext2.Add(c0, c1)        // t1.Add(c0, c1)
+	tmp = e.Ext2.Add(&z.B0, &z.B1)  // tmp.Add(&z.B0, &z.B1)
+	t1 = e.Ext2.Mul(t1, tmp)        // t1.Mul(&t1, &tmp)
+	t1 = e.Ext2.Sub(t1, a)          // t1.Sub(&t1, &a)
+	t1 = e.Ext2.Sub(t1, b)          // t1.Sub(&t1, &b)
+	return &E6{
+		B0: *t0, // z.B0.Set(&t0)
+		B1: *t1, // z.B1.Set(&t1)
+		B2: *t2, // z.B2.Set(&t2)
+	} // return z
+}
+
+func (e Ext6) MulByNonResidue(x *E6) *E6 {
+	z2, z1, z0 := &x.B1, &x.B0, &x.B2 // z.B2, z.B1, z.B0 = x.B1, x.B0, x.B2
+	z0 = e.Ext2.MulByNonResidue(z0)   // z.B0.MulByNonResidue(&z.B0)
+	return &E6{                       // return z
+		B0: *z0,
+		B1: *z1,
+		B2: *z2,
+	}
+}
+
+func (e Ext6) AssertIsEqual(x, y *E6) {
+	e.Ext2.AssertIsEqual(&x.B0, &y.B0)
+	e.Ext2.AssertIsEqual(&x.B1, &y.B1)
+	e.Ext2.AssertIsEqual(&x.B2, &y.B2)
+}
+
+func (x *E6) assign(y *bn254.E6) {
```

Comment:
Same note as for creating `E2` (i.e. instead of method on pointer receiver on E6 have `FromE6` function)